interface/wx/webview.h

```diff
@@ -459,12 +459,50 @@ class wxWebView : public wxControl
     */
     virtual void Reload(wxWebViewReloadFlags flags = wxWEBVIEW_RELOAD_DEFAULT) = 0;
 
-    /**
-        Runs the given javascript code.
-        @note When using wxWEBVIEW_BACKEND_IE you must wait for the current
+	/**
+        Gets current IE emulation level for the current program.
+        @return Current wxWebViewIEEmulationLevel.
+        @note Only avaliable on wxWEBVIEW_BACKEND_IE.
+	*/
+	virtual wxWebViewIEEmulationLevel GetEmulationLevel() = 0;
+
+	/**
+        Sets IE emulation level.
+        @param level A level from wxWebViewIEEmulationLevel enum.
+        @return True if level can be set, false if it is not.
+        @note Only avaliable on wxWEBVIEW_BACKEND_IE.
+	*/
+    virtual bool SetEmulationLevel(wxWebViewIEEmulationLevel level) = 0;
+
+    /**
+        Runs the given JavaScript. Also it does not return the result, it returns 
+        if there was an error and sets the output parameter to the result.
+        It returns strings, integers, floating point numbers, booleans and objects (as JSON).
+        @note When using wxWEBVIEW_WEBKIT (OSX), there are two limits:
+              1) Javascript allocations greater than 10MB.
+              2) Javascript that takes longer than 10 seconds to execute.
+
+              When using wxWEBVIEW_BACKEND_IE you must wait for the current
               page to finish loading before calling RunScript().
-    */
-    virtual void RunScript(const wxString& javascript) = 0;
+
+              When using wxWEBVIEW_BACKEND_IE, JSON is not available in Quirks or 
+              IE6/7 standards mode, which is unfortunately the default one for the embedded browser control, see
+              https://docs.microsoft.com/en-us/scripting/javascript/reference/json-object-javascript#requirements
+              and see here how to make a program run use "modern" modes
+              https://msdn.microsoft.com/en-us/library/ee330730(v=vs.85)#browser_emulation
+              You can use SetEmulationLevel to change emulation level, for example:
+              @code
+                  SetEmulationLevel(wxWEBVIEWIE_EMULATION_LEVEL_7_DEFAULT);
+              @endcode
+              However, if you don't want to use this, there is an implementation of JSON parser
+              on RunScript that helps to return objects.
+              Also, it is necessary a script tag inside HTML to run Javascript. We recommend a minimum DOM
+              on the HTML document.
+        @param javascript A wxString containing the Javascript.
+        @param output wxString result pointer.
```

Comment:
You should also say whether `output` can be null or not and should additionally mention that this parameter is new in wxWidgets 3.1.1 (you can't use `@since` in this context, so you'll have to spell it out).